### ISSUE TITLE
Error saving project gives me a new channel ID

### ISSUE THREAD

**Issue by poorvasingal** [code-dot-org/dance-party#195]
I wish I knew how to repro this but I occasionally get the "Error saving project" red error in the top teal bar, which then results me in getting a new channel ID. At least that's what I think is happening because I kept getting a different URL in the finish dialog after running into this error.

**Comment by joshlory** [comment]
@islemaster you saw this happen in our testing, where the user would get a channel token not actually owned by the signed-in user.  Did you ever track down root cause?

**Comment by islemaster** [comment]
No, I didn't.  The mystery to me here is why we haven't seen this all over the place with all of our project types.  Not sure why this would be dance-specific.

**Comment by joshlory** [comment]
I doubt it _is_ dance-specific.  Do we have any metrics/tracking around save failures in project levels?

**Comment by poorvasingal** [comment]
@davidsbailey - I think you might know if we ever log when users get the "Error saving project" error?

**Comment by davidsbailey** [comment]
Yep! For the project id in question, go to sqlworkbench and run a query like: `select * from analysis.events where project_id = '6qQ676YYRb1oCFZnsUjfrRtI2wqmVBoLjPPt83JHhj0';` (e.g. this would be from a url like https://studio.code.org/projects/dancelab/6qQ676YYRb1oCFZnsUjfrRtI2wqmVBoLjPPt83JHhj0/)

The fact that we are seeing this more often now probably has to do with the project being owned by non-signed-in users. That's just a guess, but it seems like the biggest difference between dance and other projects.

**Comment by davidsbailey** [comment]
From context I assume this is happening on the free play level, https://studio.code.org/s/dance/stage/1/puzzle/13 . 

1. one possible scenario is that you did work on this level while signed in, then signed out in another window, so we are unable to save the work associated with your account. Red error bar, logging a warning, and page reload is expected behavior. In this case, you can find your work by signing back in.

2. Another scenario is the inverse, that you were signed out, worked on dance free play, then signed in. Again Red error bar, logged error, and page reload are expected. In this case, if you sign out, you will see your work that disappeared after you signed in.

3. third scenario is that you opened https://studio.code.org/s/dance/stage/1/puzzle/13 in two browser windows and made changes in both. when we detect this, we again show red, log error, and reload.

After looking at Poorva's browser, I unfortunately don't see any of the error logging I would expect to see for any of the above scenarios. Current hypothesis is that this could have been b/c poorva was using incognito, so there's no way for me to track down the project id to look for errors on it.

If anyone sees these errors in the future, please be sure to note (1) the share url and (2) the account you are signed in as and I'll investigate further. otherwise, I believe this is all working as intended.

**Comment by davidsbailey** [comment]
**TL;DR:** this seems to be **working as .intended**.

**Comment by poorvasingal** [comment]
Yep, I wouldn't investigate this further esp. because I haven't seen this error happening since.  

@boatmarina - I think this issue can be closed.
